### PR TITLE
add new target Stellar H7 V2

### PR DESCRIPTION
**Pull request by cvetaevvitaliy**
add new target Stellar H7 V2
The scheme was sent to StingBee Discord channel

### CODE REVIEW

File: configs/STELLARH7V2/config.h [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+#define ADC3_DMA_OPT        9
+#define TIMUP1_DMA_OPT      0
+#define TIMUP2_DMA_OPT      0
+#define TIMUP3_DMA_OPT      2
+#define TIMUP4_DMA_OPT      1
+#define TIMUP5_DMA_OPT      0
+#define TIMUP8_DMA_OPT      0
+
+#define BARO_I2C_INSTANCE (I2CDEV_2)
+#define MAG_I2C_INSTANCE (I2CDEV_2)
```
Comment (haslinghuis) [inline configs/STELLARH7V2/config.h:128]:
```suggestion
#define BARO_I2C_INSTANCE I2CDEV_2
#define MAG_I2C_INSTANCE I2CDEV_2
```

File: configs/STELLARH7V2/config.h [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define FC_TARGET_MCU     STM32H743
+
+#define BOARD_NAME        STELLARH7V2
```
[the remark this answers is not in the document]
Comment (cvetaevvitaliy) [reply configs/STELLARH7V2/config.h:26]: STELLARH7V2 is needed because it will be STELLARH7, it is at the debugging stage, a mini version for racing

**Review by haslinghuis — APPROVED** [review]

File: configs/STELLARH7V2/config.h [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+
+#define ADC1_DMA_OPT        8
+#define ADC3_DMA_OPT        9
+#define TIMUP1_DMA_OPT      0
+#define TIMUP2_DMA_OPT      0
+#define TIMUP3_DMA_OPT      2
+#define TIMUP4_DMA_OPT      1
+#define TIMUP5_DMA_OPT      0
+#define TIMUP8_DMA_OPT      0
```
Comment (ot0tot) [inline configs/STELLARH7V2/config.h:125]:
```suggestion
#define TIMER_PIN_MAPPING \
    TIMER_PIN_MAP( 0, PB0 , 2,  0) \
    TIMER_PIN_MAP( 1, PB1 , 2,  1) \
    TIMER_PIN_MAP( 2, PA0 , 2,  2) \
    TIMER_PIN_MAP( 3, PA1 , 2,  3) \
    TIMER_PIN_MAP( 4, PA2 , 2,  4) \
    TIMER_PIN_MAP( 5, PA3 , 2,  5) \
    TIMER_PIN_MAP( 6, PD12, 1,  6) \
    TIMER_PIN_MAP( 7, PD13, 1,  7) \
    TIMER_PIN_MAP( 8, PD14, 1, -1) \
    TIMER_PIN_MAP( 9, PD15, 1, -1) \
    TIMER_PIN_MAP(10, PA15, 1, -1) 



#define ADC1_DMA_OPT        8
#define ADC3_DMA_OPT        9
#define TIMUP1_DMA_OPT      0
#define TIMUP2_DMA_OPT      0
#define TIMUP3_DMA_OPT      10
#define TIMUP4_DMA_OPT      11
#define TIMUP5_DMA_OPT      12
#define TIMUP8_DMA_OPT      0
```

File: configs/STELLARH7V2/config.h [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+#define BARO_I2C_INSTANCE I2CDEV_2
+#define MAG_I2C_INSTANCE I2CDEV_2
+
+#define DEFAULT_RX_FEATURE FEATURE_RX_SERIAL
+#define SERIALRX_UART SERIAL_PORT_USART2
+#define SERIALRX_PROVIDER SERIALRX_CRSF
```
Comment (ot0tot) [inline configs/STELLARH7V2/config.h:132]: Does this FC include a receiver? Why include this in the define?

Comment (haslinghuis) [reply configs/STELLARH7V2/config.h:132]: Like TBS it has space for mounting CRSF on top of the FC.

Comment (cvetaevvitaliy) [reply configs/STELLARH7V2/config.h:132]: FC has place for RX
Screenshot 2024-12-06 172056

Comment (ot0tot) [reply configs/STELLARH7V2/config.h:132]: Pads are labeled TX/RX4, and serialRX is set to UART2

Comment (cvetaevvitaliy) [reply configs/STELLARH7V2/config.h:132]: Thanks, typo

Comment (haslinghuis) [reply configs/STELLARH7V2/config.h:132]: On the silk ?

Comment (cvetaevvitaliy) [reply configs/STELLARH7V2/config.h:132]: > On the silk ?

No, in config.

I'm going home from the office now and I'll make commit.

File: configs/STELLARH7V2/config.h [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+#define TIMUP3_DMA_OPT      10
+#define TIMUP4_DMA_OPT      11
+#define TIMUP5_DMA_OPT      12
+#define TIMUP8_DMA_OPT      0
+
+#define BARO_I2C_INSTANCE I2CDEV_2
+#define MAG_I2C_INSTANCE I2CDEV_2
+
+#define DEFAULT_RX_FEATURE FEATURE_RX_SERIAL
+#define SERIALRX_UART SERIAL_PORT_USART2
```
Comment (haslinghuis) [inline configs/STELLARH7V2/config.h:131]:
```suggestion
#define SERIALRX_UART SERIAL_PORT_UART4
```

File: configs/STELLARH7V2/config.h (hunk 4 above)
Comment (haslinghuis) [reply configs/STELLARH7V2/config.h]: Fixed

Comment (cvetaevvitaliy) [reply configs/STELLARH7V2/config.h]: Thanks

**Review by ot0tot — APPROVED** [review]

**Comment by ot0tot** [comment]
Can also add `#define BOX_USER1_NAME       "CAM 1,2"` to set the name for USER1 in the modes tab.

**Comment by cvetaevvitaliy** [comment]
> Can also add `#define BOX_USER1_NAME       "CAM 1,2"` to set the name for USER1 in the modes tab.

Yes! I forgot, I did this implementation in 4.5.x
Thanks

**Pull request merged**